src/Microsoft.AspNet.Mvc.Core/ViewDataDictionary.cs

```diff
@@ -187,16 +184,13 @@ protected ViewDataDictionary([NotNull] ViewDataDictionary source, object model,
         {
             // This is the core constructor called when Model is known.
             var modelType = GetModelType(model);
-            if (modelType == source.ModelMetadata.ModelType && model == source.ModelMetadata.Model)
+            if (modelType == source.ModelMetadata.ModelType && model == source.ModelExplorer.Model)
             {
-                // Preserve any customizations made to source.ModelMetadata if the Type that will be calculated in
-                // SetModel() and source.Model match new instance's values.
```

Comment:
In the next change there will be customizations to model metadata. Will add it back until then.